server/util/redact/redact.go

```diff
@@ -87,10 +87,10 @@ var (
 	// Here we match 20 alphanumeric characters preceded by the api key header flag
 	apiKeyHeaderPattern = regexp.MustCompile("x-buildbuddy-api-key=[[:alnum:]]{20}")
 
-	// Here we match 20 alphanum chars at the start of a line or anywhere in the
-	// line, preceded by a non-alphanum char (to ensure the match is exactly 20
-	// alphanum chars long), followed by an @ symbol.
-	apiKeyAtPattern = regexp.MustCompile("(^|[^[:alnum:]])[[:alnum:]]{20}@")
+	// Match sequences that look like API keys immediately followed '@',
+	// to account for patterns like "grpc://$API_KEY@app.buildbuddy.io"
+	// or "bes_backend=$API_KEY@domain.com".
+	apiKeyAtPattern = regexp.MustCompile("(^|[/=]])[[:alnum:]]{20}@")
```

Comment:
Doesn't this have an additional `]`? Pretty sure it should be 
```suggestion
	apiKeyAtPattern = regexp.MustCompile("(^|[/=])[[:alnum:]]{20}@")
```
, no?

Reply:
Haha, it does. Must have been excluding `/`, `=`, and `]`?

Reply:
Nah, it would have been matching either start of line (`^`), or matching a slash or equals (`[/=]`) followed by a `]`.

Do we have a test for this? We should probably test regexes, especially ones with security implications.

Reply:
`redactAPIKeys(...)` is the only function that uses `apiKeyAtPattern`. There are two callers:
1. `RedactText(...)`. I've added test cases for the `RedactText(...)` tests.
2. `StreamingRedactor.RedactAPIKeysWithSlowRegexp(...)`. There are no tests of this path, but it is called from `build_event_handler`. Let me add tests for this path as well.

Reply:
Awesome, thanks!